llms/mlx_lm/models/llama.py

```diff
@@ -30,12 +32,128 @@ def __post_init__(self):
             self.num_key_value_heads = self.num_attention_heads
 
         if self.rope_scaling:
-            required_keys = {"factor", "type"}
-            if not all(key in self.rope_scaling for key in required_keys):
-                raise ValueError(f"rope_scaling must contain keys {required_keys}")
+            if not any(key in self.rope_scaling for key in ["type", "rope_type"]):
+                raise ValueError(
+                    f"rope_scaling must contain either 'type' or 'rope_type'"
+                )
```

Comment:
Nit could simplify:

- Delete this
- Add a check below to make sure `rope_type is not None` and throw with the same message if it is.

Reply:
Fixed.